_posts/2020-01-21-fault-tolerance-4-0.adoc

```diff
@@ -0,0 +1,147 @@
+:page-layout: post
+:page-title: Fault Tolerance 4.0
+:page-synopsis: Why we reimplemented SmallRye Fault Tolerance from scratch and how does it look like?
+:page-tags: [announcement, microprofile]
+:page-date: 2020-01-21 18:00:00.000 +0100
+:page-author: lthon
+:uri-smallrye-ft: https://github.com/smallrye/smallrye-fault-tolerance
+:uri-microprofile-ft: https://github.com/eclipse/microprofile-fault-tolerance
+:uri-microprofile-context-propagation: https://github.com/eclipse/microprofile-context-propagation
+:uri-microprofile-opentracing: https://github.com/eclipse/microprofile-opentracing
+:uri-hystrix: https://github.com/Netflix/Hystrix
+
+= Fault Tolerance 4.0
+
+{uri-smallrye-ft}[SmallRye Fault Tolerance] is our implementation of {uri-microprofile-ft}[Eclipse MicroProfile Fault Tolerance].
+It has been originally based on {uri-hystrix}[Hystrix], the Netflix library for latency and fault tolerance for distributed systems.
+Lately, we realized that Hystrix is not the best fit for implementing MicroProfile Fault Tolerance, because Hystrix meshes all fault tolerance concerns into one, while MicroProfile Fault Tolerance describes mostly a layered architecture.
+The MicroProfile Fault Tolerance specification also requires certain features that Hystrix intentionally doesn't provide; most importantly, the ability to interrupt threads that Hystrix itself didn't create.
+
+We have been talking on and off about removing Hystrix and using our own implementation of core fault tolerance strategies, one that would be better suited as a base for implementing MicroProfile Fault Tolerance, but only a few months ago did we finally start working on it.
+Thanks to heroic work of Michał Szynkiewicz, the new implementation didn't take long to pass the entire MicroProfile Fault Tolerance TCK.
```

Comment:
Not critical, but maybe a link to Michal's GH with his name?

Reply:
Fixed, thanks!